storage/innobase/include/dict0crea.inl

```diff
@@ -54,7 +54,7 @@ dict_create_add_foreign_id(
 		if (dict_table_t::is_temporary_name(name)) {
 
 			/* no overflow if number < 1e13 */
-			sprintf(id, "%s_ibfk_%lu", name,
+			snprintf(id, sizeof(id), "%s_ibfk_%lu", name,
```

Comment:
namelen + 20 is the length

Reply:
Instead of `sizeof(id)` -> `sizeof(nameless + 20)`